### ISSUE TITLE
Preload lang chunks

### ISSUE THREAD

**Issue by jeanphilippeds** [nuxt-community/nuxt-i18n#486]
### What problem does this feature solve?

Hello,

I'm working on a 30-lang website and it would be great to preload some lang chunks.

Ex:
- On http://website.com/it/ ('it' and 'en' chunks are preloaded, as 'en' is the fallback language)
- On http://website.com/fr/ ('fr' and 'en' chunks are preloaded)

Btw, is it possible to do so today?

Thanks,

<div align="right"><sub><em>This feature request is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt-i18n/issues/c319">#c319</a>)</em></sub></div>

**Comment by rchl** [comment]
This is how it works right now with `lazyLoad` enabled. Do you want to customize it more or don't know about existing behavior?

**Comment by jeanphilippeds** [comment]
Indeed, it's linked to the `lazyload` feature. I can't disable it, otherwise I would bundle translations for 30 countries, having a huge impact on the bundle size.

**Comment by rchl** [comment]
By preload you mean use the preload meta tag?

**Comment by jeanphilippeds** [comment]
Exactly!

**Comment by jeanphilippeds** [comment]
I managed to do it, although I'm not really proud of the solution:

```javascript
if (process.client){
    import('~/path/to/translations/messages.en_GB.yml')
    import(`~/path/to/translations/messages.${__NUXT__.state.layout.locale}.yml`)
}
```

**Comment by rchl** [comment]
I thought I've understood what you were requesting but your last comment has confused me again.

Apparently you are not talking about `preload` tags (https://developer.mozilla.org/en-US/docs/Web/HTML/Preloading_content).

Lazy loading itself is loading all messages for the current locale and fallback locale so I'm not sure what are you missing exactly and how importing those yml files helps you.

**Comment by jeanphilippeds** [comment]
I'll try to be clear enough:

Conditions:
- I'm using Nuxt-i18n
- `lazy` feature is activated

What is the problem?
- On client side, when JS is executed, lang files are loaded sequentially:
  - First, the browser gets the fallback lang file (in my case: `en_GB`)
  - Then, it fetches the current locale lang file (let's say `fr_FR`)
It has an impact on my website performance as the user is waiting for the server to answer, twice.

What solutions?
- Use a preload tag to get those 2 files sooner. But I did not manage to configure it properly.
- Use webpack function `import()` to:
   - Get the lang files when the JS execution starts.
   - The browser caches it so it's already loaded when it's needed later.

**Comment by rchl** [comment]
Now I understand, thanks :)
Since fallback is always required, it would make sense to have it in the main bundle already.
For the current language, I'd try to add preload tags.

No idea how to do those things yet but will investigate at some point. :)

**Comment by tvld** [comment]
As far as I can see, 'fallback' does not work at all. When I set language to ``en`` all is fine. But when I switch to other (subset), it will show me: 
```
[vue-i18n] Value of key 'home.title' is not a string! vue-i18n.esm.js:33
[vue-i18n] Cannot translate the value of keypath 'home.pay_off.title'. Use the value of keypath as default.
```

**Comment by tvld** [comment]
Found some solution in https://github.com/nuxt-community/nuxt-i18n/issues/34

**Comment by rchl** [comment]
@tvld Please don't post the same issue in multiple issues. It almost looks like spam at this point.

I thought that your issue was to do with missing `vueI18n.fallbackLocale` but since you are still talking about it being broken, then please create reproducible case on codesandbox and create a new issue for it. I can't help you otherwise and repeating yourself in multiple issues won't help either.

And since those comments are offtopic for this issue (which is about preloading files that are already loaded), I'm gonna mark them as such now.

**Comment by tvld** [comment]
@rchl  yeah, sorry was ill-attempt to find commonality between issues, should have deleted myself.

**Comment by tvld** [comment]
It's still a relevant issue. To avoid delay or screen flash for users that need the non-english, preload of the native might be preferable first... ))?

**Comment by rchl** [comment]
With #823 we'll load translations on the server and pass that to the client so extra network requests (combined with #820) will be completely avoided.

That means that there is no need to add extra preload tags.

**Comment by rchl** [comment]
Released in [v6.13.3](https://github.com/nuxt-community/i18n-module/releases/tag/v6.13.3)
